app/src/main/java/com/example/taskyy/ui/screens/AgendaScreen.kt

```diff
@@ -140,6 +156,31 @@ fun AgendaScreen(state: AgendaState, onEvent: (AgendaEvent) -> Unit, timeDateSta
     )
 }
 
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun StartWorkManager() {
+    val context = LocalContext.current
+    LaunchedEffect(key1 = Unit) {
+        val constraints = Constraints.Builder()
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+        val workRequest = PeriodicWorkRequestBuilder<AgendaItemWorker>(
+            repeatInterval = 2,
+            repeatIntervalTimeUnit = TimeUnit.MINUTES
+        ).setBackoffCriteria(
+            backoffPolicy = BackoffPolicy.LINEAR,
+            duration = Duration.ofSeconds(15)
+        ).setConstraints(constraints)
+            .build()
+        val workManager = WorkManager.getInstance(context)
+        workManager.enqueueUniquePeriodicWork(
+            "AgendaItemWorker",
+            ExistingPeriodicWorkPolicy.KEEP,
+            workRequest
+        )
+    }
+}
```

Comment:
This does not need to be a composable. The UI should not be responsible for handling background syncs and data related stuff. Instead, you can put this in a class in data and have an abstraction in domain which lets you schedule such a worker